loader/loader.go

```diff
@@ -0,0 +1,281 @@
+	return runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObject.UnstructuredContent(), crd)
+}
+
+// getObject loads an object from the cluster. This is a generic function that requires the object to be passed as an
+// argument. The object is modified during the invocation, but it's returned as well to simplify the implementation of
+// functions using this function.
+func getObject[T any](name, namespace string, cli client.Client, ctx context.Context, object *T) (*T, error) {
+	unstructuredObject, err := getUnstructuredObject(object)
+	if err != nil {
+		return nil, err
+	}
+
+	err = cli.Get(ctx, types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, unstructuredObject)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return object, convertUnstructuredObject(unstructuredObject, object)
+}
+
+// getUnstructuredObject returns an unstructured object containing the same data as the object passed as an argument.
+func getUnstructuredObject[T any](object T) (*unstructured.Unstructured, error) {
+	unstructuredObject := &unstructured.Unstructured{}
+	return unstructuredObject, scheme.Scheme.Convert(object, unstructuredObject, nil)
+}
+
+// GetActiveReleasePlanAdmission returns the ReleasePlanAdmission targeted by the given ReleasePlan.
+// Only ReleasePlanAdmissions with the 'auto-release' label set to true (or missing the label, which is
+// treated the same as having the label and it being set to true) will be searched for. If a matching
+// ReleasePlanAdmission is not found or the List operation fails, an error will be returned.
+func GetActiveReleasePlanAdmission(releasePlan *v1alpha1.ReleasePlan, cli client.Client, ctx context.Context) (*v1alpha1.ReleasePlanAdmission, error) {
+	releasePlanAdmissions := &v1alpha1.ReleasePlanAdmissionList{}
+	opts := []client.ListOption{
+		client.InNamespace(releasePlan.Spec.Target),
+		client.MatchingFields{"spec.origin": releasePlan.Namespace},
+	}
+
+	err := cli.List(ctx, releasePlanAdmissions, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	activeReleasePlanAdmissionFound := false
+
+	for _, releasePlanAdmission := range releasePlanAdmissions.Items {
+		if releasePlanAdmission.Spec.Application == releasePlan.Spec.Application {
+			labelValue, found := releasePlanAdmission.GetLabels()[v1alpha1.AutoReleaseLabel]
+			if found && labelValue == "false" {
+				return nil, fmt.Errorf("found ReleasePlanAdmission '%s' with auto-release label set to false",
+					releasePlanAdmission.Name)
+			}
+			activeReleasePlanAdmissionFound = true
+		}
+	}
+
+	if !activeReleasePlanAdmissionFound {
+		return nil, fmt.Errorf("no ReleasePlanAdmission found in the target (%+v) for application '%s'",
+			releasePlan.Spec.Target, releasePlan.Spec.Application)
+	}
+
+	return &releasePlanAdmissions.Items[0], nil
+}
+
+// GetActiveReleasePlanAdmissionFromRelease returns the ReleasePlanAdmission targeted by the ReleasePlan referenced by
```

Comment:
docstring was not updated for this function. I don't know why we want it

Reply:
I don't think we need all the details about the label and list operator here. They are not things in this function, but in the functions called by this function

Reply:
I think we should keep it. From the caller point of view, they don't have any idea what we do inside this function. We would be hiding the behavior.

Reply:
I suppose..